pulsar-broker/src/main/java/org/apache/pulsar/broker/service/plugin/EntryFilterProvider.java

```diff
@@ -37,74 +43,78 @@
 import org.apache.pulsar.common.util.ObjectMapperFactory;
 
 @Slf4j
-public class EntryFilterProvider {
+public class EntryFilterProvider implements AutoCloseable {
 
     @VisibleForTesting
     static final String ENTRY_FILTER_DEFINITION_FILE = "entry_filter";
 
-    /**
-     * create entry filter instance.
-     */
-    public static ImmutableMap<String, EntryFilterWithClassLoader> createEntryFilters(ServiceConfiguration conf,
-                                                                                      EntryFilters entryFilters)
+    private final ServiceConfiguration serviceConfiguration;
+    private Map<String, EntryFilterMetaData> definitions;
+    private Map<String, NarClassLoader> cachedClassLoaders;
+    private final List<EntryFilterWithClassLoader> brokerEntryFilters;
+
+    public EntryFilterProvider(ServiceConfiguration conf) throws IOException {
+        this.serviceConfiguration = conf;
+        initialize();
+        if (!conf.getEntryFilterNames().isEmpty()) {
+            brokerEntryFilters = loadEntryFilters(conf.getEntryFilterNames());
+        } else {
+            brokerEntryFilters = Collections.emptyList();
+        }
+    }
+
+    public List<EntryFilterWithClassLoader> loadEntryFiltersForPolicy(EntryFilters policy)
```

Comment:
In the original implementation, `EntryFilterWithClassLoader` is unnecessarily used as a top level concept. I think that this is a bad approach. It's sufficient to use `EntryFilter`. 

Please refactor the solution to hide `EntryFilterWithClassLoader`. It's possible that it could be removed completely.

Reply:
Unfortunately we need to keep a reference to the NarClassLoader, this is a pattern we have in other plugins that can be loaded from a Nar file